test/unit/app/models/reshuffle_mode_test.rb

```diff
@@ -0,0 +1,72 @@
+require "test_helper"
+
+class ReshuffleModeTest < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+
+  class ClassThatIncludesReshuffleMode < ApplicationRecord
+    self.table_name = "organisations"
+    include ReshuffleMode
+  end
+
+  describe "#reshuffle_in_progress?" do
+    it "returns false when reshuffle mode is switched off" do
+      assert_not ClassThatIncludesReshuffleMode.new.reshuffle_in_progress?
+    end
+
+    it "returns true when reshuffle mode is switched on" do
+      create(:sitewide_setting, key: :minister_reshuffle_mode, on: true)
+
+      assert ClassThatIncludesReshuffleMode.new.reshuffle_in_progress?
+    end
+  end
+
+  describe "#republish_ministerial_pages_to_publishing_api" do
+    it "republishes 'How Government Works' and 'Ministers Index' pages" do
+      PresentPageToPublishingApiWorker.expects(:perform_async).with("PublishingApi::HowGovernmentWorksPresenter", true).once
+      PresentPageToPublishingApiWorker.expects(:perform_async).with("PublishingApi::MinistersIndexPresenter", true).once
+
+      ClassThatIncludesReshuffleMode.new.republish_ministerial_pages_to_publishing_api
+    end
+
+    it "only saves to the draft stack when reshuffle mode is switched on" do
+      PresentPageToPublishingApiWorker.expects(:perform_async).with("PublishingApi::HowGovernmentWorksPresenter", false).once
+      PresentPageToPublishingApiWorker.expects(:perform_async).with("PublishingApi::MinistersIndexPresenter", false).once
+
+      reshuffle_mode = ClassThatIncludesReshuffleMode.new
+      reshuffle_mode.stubs(:reshuffle_in_progress?).returns(true)
```

Comment:
Bit of a nitpick, but I don't love stubbing a method on the public interface of the class under test. I know it's covered above, so I don't feel strongly about it, it's more of an instinctive "don't mock things you don't have to" reaction even though it's probably fine in this case

Reply:
Fair point, though I like that we've already testing `reshuffle_in_progress?` above, so the rest of the tests don't have to worry about implementation detail (and can be stubbed in a quick, easy, readable way). It means we can change the implementation of `reshuffle_in_progress?` (e.g. maybe one day it will be decided by both a sitewide setting _and_ something else) and we won't have to update these tests.